cmd/legacy/heartbeat/params.go

```diff
@@ -344,128 +334,97 @@ func parseExtraHeartbeat(data string) ([]heartbeat.Heartbeat, error) {
 			}
 		}
 
-		var timestamp float64
-
-		switch {
-		case h.Time != 0:
-			timestamp = h.Time
-		case h.Timestamp != 0:
-			timestamp = h.Timestamp
-		default:
-			return nil, fmt.Errorf("skipping extra heartbeat, as no valid timestamp was defined")
-		}
-
-		heartbeats = append(heartbeats, heartbeat.Heartbeat{
-			Category:          h.Category,
-			CursorPosition:    h.CursorPosition,
-			Entity:            h.Entity,
-			EntityType:        entityType,
-			IsWrite:           h.IsWrite,
-			Language:          h.Language,
-			LanguageAlternate: h.LanguageAlternate,
-			LineNumber:        h.LineNumber,
-			Lines:             h.Lines,
-			ProjectAlternate:  h.ProjectAlternate,
-			ProjectOverride:   h.Project,
-			Time:              timestamp,
-		})
-	}
-
-	return heartbeats, nil
-}
-
-func parseExtraHeartbeatWithStringValues(data string) ([]heartbeat.Heartbeat, error) {
-	var incoming []struct {
-		Category          heartbeat.Category `json:"category"`
-		CursorPosition    *string            `json:"cursorpos"`
-		Entity            string             `json:"entity"`
-		EntityType        string             `json:"entity_type"`
-		Type              string             `json:"type"`
-		IsWrite           *bool              `json:"is_write"`
-		Language          *string            `json:"language"`
-		LanguageAlternate string             `json:"alternate_language"`
-		LineNumber        *string            `json:"lineno"`
-		Lines             *string            `json:"lines"`
-		Time              *string            `json:"time"`
-		Project           string             `json:"project"`
-		ProjectAlternate  string             `json:"alternate_project"`
-		Timestamp         *string            `json:"timestamp"`
-	}
-
-	err := json.Unmarshal([]byte(data), &incoming)
-	if err != nil {
-		return nil, fmt.Errorf("failed to json decode from data %q: %s", string(data), err)
-	}
-
-	var heartbeats []heartbeat.Heartbeat
-
-	for _, h := range incoming {
 		var cursorPosition *int
 
-		if h.CursorPosition != nil {
-			parsed, err := strconv.Atoi(*h.CursorPosition)
+		switch cursorPositionVal := h.CursorPosition.(type) {
+		case float64:
```

Comment:
I guess we should add a case for `int` type as well. This way, we support `float64`, `int` and `string`. This applies for `lineno` and `lines` values as well.